src/types/verification_modes.ts

```diff
@@ -0,0 +1,65 @@
+import { identity } from 'lodash'
```

Comment:
i think you are overcomplicating this. why can't we just use type-fest's `Opaque` type instead of a relatively unused and not-updated library?

to create an opaque type, we just need a creator method, e.g.

```ts
const createEmail = (email: string): Email => {
  // Can even add validation here and throw error or something
  return email as Email;
}
```

you really don't need something so complex like assert, etc.

everything else should stick to the same doesn't it? Only functions that absolutely need the correct type to declare it

Reply:
the reason why we can't use the `Opaque` type is because it's meant as a black box - once we use it, there's no way for us to extract the type out from `Opaque` because it's set as `Type & {readonly __opaque__: Token}` where there isn't a witness to the base type. 

wrt `assert` and `parse` i just wanted to add utility methods for ppl to use AHAH so that the api is sensible and easy to use

Reply:
we should not need to extract the type out. any arguments of 'string' will be able to be used since Opaque is a superset.